enhancements/network/user-defined-network-segmentation.md

```diff
@@ -413,6 +519,203 @@ guarantee uniqueness and eliminates the ability to falsify access.
+1. Check the owner reference match the request CR UID.
+2. OVN-K user-defined-network finalizer exist.
+3. NAD spec correspond to desired spec.
+In case one of the following checks fails, the controller should reconcile the request and 
+this state in the CR status network is not ready.
+
+In scenario of cluster-scope CRD request, the controller should do best effort to create the desired NAD in each
+specified namespace.
+In case on or more NAD creation fails (e.g.: due to namespace not exist), the controller should continue to the 
+next namespace.
+When finished, reflect in the status all namespaces where NAD creation failed to create.
+
+##### General flow
+1. On namespaced-scope CRD creation:
+  1.1. Validates the CRD spec. 
+  1.2. Generate NAD manifest from the CRD spec.
+  1.3. Check no the desired NAD is not created yet, if so, create the NAD and return.
+  1.4. Otherwise, verify the existing NAD correspond to desired spec, if so return.
+  1.5. In case the NAD is malformed raise an error and return.
+  1.6. Update the status as follows:
+   1.6.1. Reflect namespaces where network is available.
+   1.6.2. Reflect the reconciliation errors, which namespaces failed and the reason.
+
+2. on cluster-scope CRD creation:
+   2.1. Validates the CRD spec.
+   2.2. Generate NAD manifest from the CRD spec.
+   2.3. For each namespace specified in the spec:
+       2.3.1. Check no the desired NAD is not created yet, if so, create the NAD and continue. 
+       2.3.2. Otherwise, verify the existing NAD correspond to desired spec, if so continue.
+       2.3.3. In case the NAD is malformed raise an error and continue.
+   2.4. Update the status as follows:
+       2.4.1. Reflect namespaces where network is available.
+       2.4.1. Reflect the reconciliation errors, which namespaces failed and the reason.
+
+3. On namespace-scope CRD deletion:
+   3.1. If no NAD exist, return.
+   3.1. In case a NAD exist, ensure no pod specifying the network.
+     3.1.1. In case no pod specifying the network, remove the finalizer, allowing the cluster garbage collector dispose the NAD object.
+     3.1.2. Otherwise, reflect in the status the network is being deleted because its in use.
+
+4. On cluster-scope CRD deletion:
+  4.1. For each namespace specified in the spec:
+   4.1. In case a NAD exist, ensure no pod specifying the network.
+   4.1.1. In case no pod specifying the network, remove the finalizer, allowing the cluster garbage collector dispose the NAD object.
+   4.1.2. Otherwise, reflect in the status the network is cannot be deleted because its in use.
+
+##### NetworkAttachmentDefinition rendering
+The underlying OVN network ID (network-name) is represented by the NAD CNI's network config `name` field, and must be unique.
+
+The network-name is generated by the controller and should not be exposed for modification.
+
+Having the network-name unique and non-configurable, avoid the risk where a malicious entity could guess other networks name, 
```

Comment:
your CRD controller just has to figure out these namespaces are **not** a part of the selector on the cluster-wide network CRD. And reject it with an error. 

I think that provides better user experience.

Reply:
In the strategy I outlined in https://github.com/openshift/enhancements/pull/1638#issuecomment-2195722725

this is not possible. The namespaced scope CRD always has its network name prefixed with its namespace name. Therefore even if the attacker guessed the name of another namespace's network, or that of a cluster scoped network, he/she cannot connect to it because the network name will always be prefixed with the attacker's namespace name.